tests/test_init.py

```diff
@@ -0,0 +1,41 @@
+"""Test imports can happen from top-level."""
+
+import pytest
+
+import propcache
+from propcache import _helpers
+
+
+def test_api_at_top_level():
+    """Verify the public API is accessible at top-level."""
+    assert propcache.cached_property is not None
+    assert propcache.under_cached_property is not None
+    assert propcache.cached_property is _helpers.cached_property
+    assert propcache.under_cached_property is _helpers.under_cached_property
+
+
+def test_public_api_is_in_dir():
+    """Verify the public API is accessible in the module's dir()."""
+    assert "cached_property" in dir(propcache)
+    assert "under_cached_property" in dir(propcache)
+
+
+def test_public_api_is_in_all():
+    """Verify the public API is accessible in the module's __all__."""
+    assert "cached_property" in propcache.__all__
+    assert "under_cached_property" in propcache.__all__
```

Comment:
It seems to me that this is really just one repeated check so I'd just parametrize it:
```suggestion
@pytest.mark.parametrize(
    'prop_name',
    ('cached_property', 'under_cached_property'),
)
@pytest.mark.parametrize(
    'api_list',
    (dir(propcache), propcache.__all__),
    ids=('dir', '__all__')
)
def test_public_api_is_in_inspectable_object_lists(prop_name, api_list):
    """Verify the public API is discoverable programmatically.

    This checks for presence of known public decorators module's
    ``__all__`` and ``dir()``.
    """
    assert prop_name in api_list
```